include/filters/FilterBase.h

```diff
@@ -0,0 +1,77 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2021 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#pragma once
+
+#include "MooseObject.h"
+
+#include "openmc/tallies/filter.h"
+
+/// Forward declarations.
+class OpenMCCellAverageProblem;
+
+/**
+ * A class which provides a thin wrapper around an OpenMC Filter for
+ * the purposes of adding a tally via the MOOSE input syntax. This
+ * class does not need init / reset functions as non-spatial tallies
+ * don't change during simulation execution (at the moment).
+ *
+ * The filter system should NOT wrap spatial filters as the problem
```

Comment:
What are your thoughts on supporting ALL of the OpenMC filters, but requesting the user to create them from the Python API, and refer to them using a Cardinal class which gets passed an `int` to refer to the filter ID? (or something along those lines).

Reply:
I think it's a good idea and could probably be implemented in a class called `ArbitraryFilter` (or something similar). I'd want to put some padding on using an arbitrary filter with a few `mooseWarnings` and `mooseErrors` letting the user know that they cannot use local normalization or tally sum checking with an arbitrary filter;  a sum over all bins for some of the tally filter options (i.e. spherical harmonics filters, Legendre filters) don't make sense as you're summing function expansion coefficients (which are not guaranteed to equal the global tally sum).

Reply:
right, there are downsides to having a fully opaque implementation, in terms of normalization -- but it could automatically support all of the filters, and we can continue to have those which we use frequently in Cardinal (like the ones you've already added) as thin wrappers.

Reply:
My solution to the normalization problem was to have `FromXMLFilter` (the new filter filter that pulls from `tallies.xml`) throw an error if the filter at `id` is a functional expansion filter. The user can override this behaviour, but they're warned against it and informed that they would need to manually normalize all Cardinal tallies which use the functional expansion filter during post-processing if they do.